libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -247,27 +253,27 @@ class MapboxTripSession(
     }
 
     private fun updateRawLocation(rawLocation: Location) {
+        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
         ioJobController.scope.launch {
-            rawLocation.let {
-                navigator.updateLocation(it)
-            }
+            navigator.updateLocation(rawLocation)
+            lastLocationUpdateTimeMillis = Date().time
+            statusUpdateTimer.stopJobs()
         }
-        locationObservers.forEach { it.onRawLocationChanged(rawLocation) }
-        if (this.rawLocation == null) {
-            fireOffStatusPolling()
+        fireOffStatusPolling()
```

Comment:
Callin `fireOffStatusPolling` below this coroutine will later call `navigatorPolling` which calls `getStatus`, but there is no explicit synchronization between this `getStatus` call and the highlighted `updateLocation`. Especially since we're running a thread pool. There's no guarantee that `updateLocation` will be executed first.

If we want to ensure that `updateLocation` is executed first, we should perform polling login in the same coroutine. After that, we can start the coroutine or timer, which will periodically call `getStatus` again.